polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -603,36 +660,57 @@ impl<H: Copy> CandidateDescriptorV2<H> {
 }
 
 impl<H: Copy> CommittedCandidateReceiptV2<H> {
-	/// Checks if descriptor core index is equal to the committed core index.
-	/// Input `cores_per_para` is a claim queue snapshot at the candidate's relay parent, stored as
+	/// Performs checks on the UMP signals and returns them.
+	///
+	/// Also checks if descriptor core index is equal to the committed core index.
+	///
+	/// Params:
+	/// - `cores_per_para` is a claim queue snapshot at the candidate's relay parent, stored as
 	/// a mapping between `ParaId` and the cores assigned per depth.
-	pub fn check_core_index(
+	pub fn check_ump_signals(
```

Comment:
Nit: more concise naming. You'd expect to get some parsed output.

```suggestion
	pub fn parse_ump_signals(
```